platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/ConsumerProvidedVariantFinder.java

```diff
@@ -48,11 +52,19 @@ public class ConsumerProvidedVariantFinder {
     public ConsumerProvidedVariantFinder(
         VariantTransformRegistry variantTransforms,
         AttributesSchemaInternal schema,
-        ImmutableAttributesFactory attributesFactory
+        ImmutableAttributesFactory attributesFactory,
+        AttributeSchemaServices attributeSchemaServices
     ) {
         this.variantTransforms = variantTransforms;
         this.attributesFactory = attributesFactory;
-        this.matcher = new CachingAttributeMatcher(schema.matcher());
+        this.matcher = new CachingAttributeMatcher(() -> {
+            // TODO: This is incorrect. We fail to merge the consumer schema with the producer schema
```

Comment:
This will be resolved in the next round of changes. However, improvements to artifact transform selection were removed from this PR to reduce the size